app/controllers/api/v1/site_settings_controller.rb

```diff
@@ -3,10 +3,27 @@
 module Api
   module V1
     class SiteSettingsController < ApiController
+      def index
+        site_setting_data = {}
+        site_settings = SiteSetting.joins(:setting).select(:id, :name, :value)
+
+        site_settings.each do |setting|
+          site_setting_data[setting.name] = { value: setting.value, id: setting.id }
+        end
+
+        render_json data: site_setting_data, status: :ok
+      end
+
       # GET /api/v1/site_settings/:name
       def show
         render_json data: SettingGetter.new(setting_name: params[:name], provider: 'greenlight').call # TODO: - ahmad: fix provider
       end
+
+      def update
+        SiteSetting.find_by(setting_id: params[:settingId]).update(value: params[:settingValue].to_s)
```

Comment:
We should handle cases of not finding the record and providing invalid params while returning adequate feedbacks.

Reply:
Also, it may be a good idea to encapsulate the `data` in a root key and leave the query params **only not** encapsulated. 
So, having an `ActionController::Parameters` instance:
`{ id: '' ,search: '', sort: { column: '', direction: '' }, site_setting: { value: '' } }`
May be better to maintain then:
`{ settingId: '' , settingValue: '' ,search: '', sort_column: '', sort_direction: '' } }`

For the record, for GL3 at the moment both are valid since we don't have a lot of `params` but it's worth knowing.